services/cd-service/pkg/repository/transformer.go

```diff
@@ -365,6 +366,10 @@ func (c *CreateUndeployApplicationVersion) Transform(ctx context.Context, state
 	}
 	result := ""
 	for env := range configs {
+		err := state.checkUserPermissions(ctx, env, c.Application, auth.PermissionCreateUndeploy, c.RBACConfig)
+		if err != nil {
+			return "", err
+		}
```

Comment:
down below in DeployApplicationVersion, also pass the authentication along

Reply:
Added authentication for `DeployApplicationVersion` and a test.